bpf/bpf_lxc.c

```diff
@@ -829,17 +829,16 @@ static inline int __inline__ ipv6_policy(struct __sk_buff *skb, int ifindex, __u
 			return ret2;
 	}
 
-	/* Policy lookup is done on every packet to account for packets that
-	 * passed through the allowed consumer. */
-	/* FIXME: Add option to disable policy accounting and avoid policy
-	 * lookup if policy accounting is disabled */
 	verdict = policy_can_access(&POLICY_MAP, skb, src_label, tuple.dport,
 				    tuple.nexthdr, sizeof(tuple.saddr),
 				    &tuple.saddr);
-	if (unlikely(ret == CT_NEW)) {
-		if (verdict != TC_ACT_OK)
-			return DROP_POLICY;
 
+	/* Reply packets and related packets are allowed, all others must be
+	 * permitted by policy */
+	if (ret != CT_REPLY && ret != CT_RELATED && verdict != TC_ACT_OK)
+		return DROP_POLICY;
+
+	if (unlikely(ret == CT_NEW)) {
```

Comment:
by the way, is it really that unlikely to get new connections?

Reply:
Not really, will remove